Converted capsule installer into scenario

mbacovsky opened the pull request:
**Do Not Merge**

This PR is for testing purposes.
- requires sources with https://github.com/Katello/katello-installer/pull/290
- scratch build is here http://koji.katello.org/koji/taskinfo?taskID=416557
- added foreman-installer-katello

For testing scartch builds of kafo, kafo_wizards and foreman_installer should be used.

```
yum -y install "http://koji.katello.org/koji/getfile?taskID=412086&name=rubygem-kafo_wizards-0.0.1-1.git.0.9c80245.git.0.9c80245.el7.noarch.rpm"
yum -y install "http://koji.katello.org/koji/getfile?taskID=414397&name=rubygem-kafo-0.7.1-1.el7.noarch.rpm"
yum -y install "http://koji.katello.org/koji/getfile?taskID=416617&name=foreman-installer-1.11.0-0.develop.el7.noarch.rpm"
yum -y install "http://koji.katello.org/koji/getfile?taskID=418934&name=katello-2.5.0-4.nightly.git.6.a5f0e39.el7.noarch.rpm http://koji.katello.org/koji/getfile?taskID=418934&name=katello-service-2.5.0-4.nightly.git.6.a5f0e39.el7.noarch.rpm http://koji.katello.org/koji/getfile?taskID=418934&name=katello-common-2.5.0-4.nightly.git.6.a5f0e39.el7.noarch.rpm"
yum -y install "http://koji.katello.org/koji/getfile?taskID=419223&name=katello-installer-base-2.5.0-1.nightly.el7.noarch.rpm"
yum -y install "http://koji.katello.org/koji/getfile?taskID=419223&name=foreman-installer-capsule-2.5.0-1.nightly.el7.noarch.rpm"
yum -y install "http://koji.katello.org/koji/getfile?taskID=419223&name=foreman-installer-katello-2.5.0-1.nightly.el7.noarch.rpm"
```

For use with `katello-deploy` following `boxes.yaml` can be used:

``` yaml
scenario-katello:
  box: centos7-katello-nightly
  options: --koji-task 412086,414397,416617,419223,418934

scenario-foreman:
  box: centos7-foreman-nightly
  options: --koji-task 412086,414397,416617

#419223 - katello-installer
#418934 - katello
#416617 - foreman-installer
#414397 - rubygem-kafo
#412086 - rubygem-kafo-wizards
```

mbacovsky committed:
Fixes #10163 - Converted katello installer into scenario based installer

ehelms commented on katello-installer/katello-installer-base.spec:
```diff
@@ -46,32 +45,27 @@ A set of tools for installation of Katello and a the default Capsule.
 
-%package -n capsule-installer
-Summary:   Puppet-based installer for a Katello Capsule
+%package -n foreman-installer-capsule
```
We should set this package to obsolete the capsule-installer to prevent  this from being used still on brownfield deployments. This may cause the config files to be renamed slightly which may impact the migration.

mbacovsky replied:
Updated, new scratch build is http://koji.katello.org/koji/taskinfo?taskID=413218

mbacovsky commented:
Links to builds in the PR description were updated to latest ones.

mbacovsky commented:
Upodated, katello-devel converted. 
Scratch: http://koji.katello.org/koji/taskinfo?taskID=422886

ehelms commented on katello-installer/katello-installer-base.spec:
```diff
@@ -9,85 +9,72 @@ Summary: Puppet-based installer for the Katello and Katello Capsule
 
 BuildArch: noarch
-Conflicts: foreman-installer
 Obsoletes: katello-installer < 2.1.0
```
We could probably remove this line given the new obsoletes

mbacovsky replied:
Could we? `foreman-installer-capsule` oboletes old `katello-installer` only via this line.
This could cover situation when someone installing new capsule scenario directly over `katello-installer < 2.1.0`. Not sure if this is supported.

ehelms replied:
Ahh, so maybe we need to update this to < 3.0.0 ?

ehelms replied:
Also, you can go ahead and update the version to 3.0.0 of the spec file

ehelms commented on katello-installer/katello-installer-base.spec:
```diff
@@ -9,85 +9,72 @@ Summary: Puppet-based installer for the Katello and Katello Capsule
+%{_datadir}/foreman-installer-katello/bin
+%{_datadir}/foreman-installer-katello/checks
+%config(noreplace) %attr(600, root, root) %{_sysconfdir}/foreman/installer-scenarios.d/katello-answers.yaml
+%config(noreplace) %attr(600, root, root) %{_sysconfdir}/foreman/installer-scenarios.d/katello.yaml
```
These will need updates I believe to `/foreman-installer/`

ehelms replied:
Same for lines 36/37

ehelms commented:
I am less sure there is a benefit in having a 'katello-installer-base' RPM given this current idea and separation. What do you think since you have been building and playing with it?

mbacovsky commented:
katello-installer-base RPM holds all the modules, hooks and checks that are shared. It also keeps the deps on one place. So IMO it makes sense.
Another options would be to 
1. return all the scenarios to one package
2. package katello and capsule separately with own pruned sets of modules and hooks and checks

mbacovsky commented:
Updated, rebuild. I moved the checks to one shared place in `/usr/share/katello-installer-base/checks`. New 'boxes.yaml`:

``` yaml
scenario-katello:
  box: centos7-katello-nightly
  options: --skip-installer --koji-task 423261,424660,424668

scenario-capsule:
  box: centos7-katello-nightly
  options: --skip-installer --koji-task 423261,424660,424668

scenario-foreman:
  box: centos7-foreman-nightly
  options: --koji-task 423261

# 424668 - katello-installer
# 424660 - katello
# 423261 - foreman-installer
```

EDIT: fixed foreman-installer task ids

bbuckingham commented on katello-installer/katello-installer-base.spec:
```diff
@@ -9,85 +9,75 @@ Summary: Puppet-based installer for the Katello and Katello Capsule
-%package -n capsule-installer
-Summary:   Puppet-based installer for a Katello Capsule
+%package -n foreman-installer-capsule
+Summary:   Capsule scenario for Foreman installer
```
Update line 23 & 44 to be consistent (e.g. use 'foreman-installer' or 'Foreman installer' for both)

ehelms commented on katello-installer/katello-installer-base.spec:
```diff
@@ -9,85 +9,62 @@ Summary: Puppet-based installer for the Katello and Katello Capsule
 License: GPLv3+ and ASL 2.0
 URL:     http://katello.org
-Source0: https://github.com/Katello/katello-installer/archive/%{version}.tar.gz
+Source0: https://github.com/Katello/katello-installer/archive/katello-installer-%{version}.tar.gz
```
This addition to the URL results in an unresolvable URL

ehelms commented on katello-installer/katello-installer-base.spec:
```diff
@@ -9,85 +9,62 @@ Summary: Puppet-based installer for the Katello and Katello Capsule
-Conflicts: foreman-installer
 Obsoletes: katello-installer < 2.1.0
 
+Requires: foreman-installer
```
Let's set this to a minimum of 1.11.0 since it requires the scenario support

ehelms commented:
You'll have to rebase -- I merged a PR that introduced some conflicts.

mbacovsky commented:
Rebased, issues fixed. 
Capsule and Katello scenarios are now together in `foreman-installer-katello`

mbacovsky commented:
Latest boxes and scratch builds:

``` yaml
scenario-katello:
  box: centos7-katello-nightly
  options: --skip-installer --koji-task 423261,426835,426992

scenario-capsule:
  box: centos7-katello-nightly
  options: --skip-installer --koji-task 423261,426835,426992

scenario-foreman:
  box: centos7-foreman-nightly
  options: --koji-task 423261

# 426992- katello-installer
# 426835 - katello
# 423261 - foreman-installer
```

ehelms commented:
[test]

ehelms commented:
[test]

ehelms commented:
ACK - thanks @mbacovsky !
